proposals/nnnn-structured-concurrency.md

```diff
@@ -358,9 +358,20 @@ async let result = try fetchHTTPContent(of: url)
 
 Any reference to a variable that was declared in an `async let` is a suspension point, equivalent to a call to an asynchronous function, so it must occur within an `await` expression. The initializer of the `async let` is considered to be enclosed by an implicit `await` expression.
 
-If the initializer of the `async let` can throw an error, then each reference to a variable declared within that `async let` is considered to throw an error, and therefore must also be enclosed in one of `try`/`try!`/`try?`. 
+If the initializer of the `async let` can throw an error, then each reference to a variable declared within that `async let` is considered to throw an error, and therefore must also be enclosed in one of `try`/`try!`/`try?`:
 
-One of the variables for a given `async let` must be awaited at least once along all execution paths (that don't throw an error) before it goes out of scope. For example:
+```swift
+{
+  async let (yay, nay) = ("yay", throw Nay())
+  
+  await try yay // must be marked with `try`; throws Nay()
+  // implicitly guarantees `nay` also be completed at this point
+}
+```
+
+The simplest way to think about it is that anything to the right hand side of the `=` of an `async let` is initiated together (as-if in an asynchronous closure), implying that if any of the values initialized by this closure throws, all other left-hand side to-be-initialized variables must also be considered as it they had thrown that error.
+
+At least one of the variables for a given `async let` must be awaited at least once along all execution paths (that don't throw an error) before it goes out of scope. For example:
```

Comment:
Same "clause" comment. Since this was confusing, perhaps add an example with a second clause?

Reply:
Added this: 

```
There is one additional case which may need a short explanation. Multiple clauses in a single `async let` may be written like this:

```swift
{
  async
let 
    ok = "ok",
    (yay, nay) = ("yay", throw Nay())
  
  await ok
  await try yay
  // okay
}
```

In the above example one can consider each clause as it's own asynchronously initialized variable, i.e. the `ok`  is initialized on its own, and the `(yay, nay)` are initialized together as was discussed previously.
```

I'm not sure if I used the word clause correctly? Looking up the swift syntax grammar didn't really clarify which bit we mean by clause to me 🤔 

Feel free to just fix if I got it slightly wrong, thanks :)